x-pack/platform/plugins/shared/streams/server/routes/internal/attachments/route.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { z } from '@kbn/zod';
+import { STREAMS_API_PRIVILEGES } from '../../../../common/constants';
+import { createServerRoute } from '../../create_server_route';
+import type { Attachment } from '../../../lib/streams/attachments/types';
+import { ATTACHMENT_TYPES } from '../../../lib/streams/attachments/types';
+
+const attachmentTypeSchema = z.enum(ATTACHMENT_TYPES);
+
+export interface SuggestAttachmentsResponse {
+  suggestions: Attachment[];
+}
+
+const suggestAttachmentsRoute = createServerRoute({
+  endpoint: 'POST /internal/streams/{streamName}/attachments/_suggestions',
```

Comment:
most likely we don't need _suggestions at end here

Reply:
We talked on Slack and for now we'll keep `_suggestions` here. It's an internal endpoint so we can change it if needed.